### ISSUE TITLE
Issues with --encryptTo

### ISSUE THREAD

**Issue by davidchisnall** [someone1/zfsbackup-go#9]
There are several issues with `--encyptTo`.  From the documentation, I expect this to encrypt the backups using the recipient's public key, so that they can be decrypted only by the owner of the private key (and so the machine doing the backups can't read the results if it is compromised). 

The first error is superficial, the error message says:

```
You must specify a private keyring path if you provide an encryptFrom option
```

There is no `--encryptFrom` command line option.

Looking at the code https://github.com/someone1/zfsbackup-go/blob/b7d04a4f0916ea05821c058b126b431eb03b89fd/cmd/root.go#L161 it appears that this is just unconventional nomenclature (the secret and public keychains are referred to the wrong way around in the documentation).  It's somewhat concerning that a function called `GetPrivateKeyByEmail` can return a public key, though this does appear to be the case, as this simply wraps `getKeyByEmail`, which will then get either a public or private key.

**Comment by someone1** [comment]
This portion of the code morphed a few times before this project went public - clearly it needs to be cleaned up. I'll review your comments and adjust the code as necessary to clearly define intent.

The backup operation needs the ability to decrypt the manifest files for the smart backup options so it can read which backup was most recently done and select the correct backup operation and snapshot.

Things to do after briefly reviewing:
- [ ] Update error to reflect valid options
- [ ] Refactor code surrounding the keyrings and how we read them in
- [ ] Update requirements for backup operations to only require the private keyring for the "smart" backup operations.

**Comment by mschout** [comment]
I have been completely unsuccessful in getting encryption to work as well.  Some documentation about how to create the public and secret key rings would be nice.  I generated a public and private key with GnuPG and have been through numerous contortions to try to get it to work and so far have had zero luck.  I tried pointing it to the `pubring.gpg` and `secring.gpg` in my `.gnupg` directory, as well as exporting the keys to files. All with no luck.

**Comment by mschout** [comment]
Another option, simple symmetric AES encryption with a secret key would be much simpler.

**Comment by someone1** [comment]
@mschout  - Sorry you're having issues, but please don't try to hijack existing issues with support requests, you're more than welcomed to open a new issue requesting something like documentation be added for generating keypairs or adding a simpler symmetric encryption algorithm option.

When I created the repo, the link I have in the documentation actually was to where you can generate keys. The website has since changed but I believe you can still do so using the library found on [this page](https://keybase.io/kbpgp/docs/generating_a_pair). My apologies for not keeping this up-to-date or adding more explicit instructions to the documentation.

I haven't tested this but I'd assume exporting public keys from gpg like this should work:
```
# First, generate a Signing/Encryption key for user@email.com
$ gpg --full-generate-key 

# Then export the keys
$ gpg --armor --export user@email.com > pubring.asc
$ gpg --armor --export-secret-keys user@email.com > secring.asc
```

If it doesn't, please continue the discussion in a new issue and I'd be happy to assist!

**Comment by someone1** [comment]
Fixed in #26
